test/git/test_invocations.py

```diff
@@ -58,6 +58,12 @@ def make_venv(self, mode):
             self.run_in_venv(venv_dir, venv_dir,
                              'pip', 'install', '-U',
                              'pip', 'wheel', 'packaging')
+        if sys.version_info.major == 3 and sys.version_info.minor >= 12:
+            # Py 3.12 no longer default-installs setuptools in environments
+            # See PEP-632
+            self.run_in_venv(venv_dir, venv_dir,
+                             'pip', 'install', '-U',
+                             'setuptools')
```

Comment:
I'd just go ahead and merge this into the pip install above. I don't think it's important to test that you can get away without installing setuptools on other versions.

Reply:
My thinking was "Its probably important enough to make sure that the default-installed setuptools on the other versions works"  such that this installs "latest" setuptools from pypi.

Reply:
Your call. That's fine by me.